### PR TITLE
build: drop LEGACY_INCLUDE_PATH support

### PR DESCRIPTION
**Pull request by fabiobaltieri**
LEGACY_INCLUDE_PATH has been defaulting to "n" and marked as deprecated in both v3.2 and v3.3. Drop the option entirely for v3.4.

### CODE REVIEW

File: include/zephyr/zephyr.h [hunk 1]
```diff
@@ -9,8 +9,4 @@
 
 #include <zephyr/kernel.h>
 
-#ifndef CONFIG_LEGACY_INCLUDE_PATH
-#warning "<zephyr/zephyr.h> is deprecated, include <zephyr/kernel.h> instead"
-#endif
-
```
Comment (gmarull) [inline include/zephyr/zephyr.h:15]: #warning should be kept

Comment (nordicjm) [reply include/zephyr/zephyr.h:15]: If the Kconfig is removed, the project should fail to configure if the option is present, so it wouldn't get to the stage of compiling this

Comment (gmarull) [reply include/zephyr/zephyr.h:15]: We need to allow usage of `<zephyr/zephyr.h>` until deprecation period expires (until post 3.4 in this case). It was decided not to warn users if they were using legacy includes. So as this option is removed now, users need to get the warning unconditionally.

**Comment by gmarull** [comment]
@fabiobaltieri we should also remove the migration script in `scripts/utils`

File: include/zephyr/zephyr.h (hunk 1 above)
Comment (nordicjm) [reply include/zephyr/zephyr.h:15]: I forgot about that, indeed!

Comment (fabiobaltieri) [reply include/zephyr/zephyr.h:15]: Sorry I was just dropping every block used by that config, did not realize that we reused it for the zephyr.h warning. Thanks for the pointer.

**Comment by fabiobaltieri** [comment]
> @fabiobaltieri we should also remove the migration script in `scripts/utils`

Gone!

**Review by gmarull — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Review by yonsch — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
CI fix: https://github.com/zephyrproject-rtos/zephyr/pull/55007

**Pull request merged**